fuzzing/private/instrument.bzl

```diff
@@ -0,0 +1,118 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Primitives for defining fuzzing engine instrumentations."""
+
+load(
+    "//fuzzing:instrum_opts.bzl",
+    "base_opts",
+    "fuzzing_engine_opts",
+    "instrumentation_opts",
+    "sanitizer_opts",
+)
+
+def _merge_opts(left_opts, right_opts):
+    return instrumentation_opts(
+        copts = left_opts.copts + right_opts.copts,
```

Comment:
Double checking: do we need to do any deduplication when doing list concatenation, or is it OK to have the same copt specified multiple times? Ditto below

Reply:
We do not, and I think idempotent options are automatically folded. However, their order matters, e.g., an `-fsanitize=` option can be followed by an `-fno-sanitize=` option that can undo the effect of the former. So I clarified in the docstring below that the `extra_*` attributes are appended to the computed set of instrumentation options.

Reply:
SGTM, thanks.